ludwig/encoders/image_encoders.py

```diff
@@ -296,3 +296,88 @@ def input_shape(self) -> torch.Size:
     @property
     def output_shape(self) -> torch.Size:
         return self._output_shape
+
+
+@register(name='vit')
+class ViTEncoder(ImageEncoder):
+    def __init__(
+            self,
+            img_height: int,
+            img_width: int = None,
+            in_channels: int = 3,
+            use_pretrained: bool = True,
+            pretrained_model: str = 'google/vit-base-patch16-224',
+            hidden_size: int = 768,
+            num_hidden_layers: int = 12,
+            num_attention_heads: int = 12,
+            intermediate_size: int = 3072,
+            hidden_act: str = 'gelu',
+            hidden_dropout_prob: float = 0.1,
+            attention_probs_dropout_prob: float = 0.1,
+            initializer_range: float = 0.02,
+            layer_norm_eps: float = 1e-12,
+            gradient_checkpointing: bool = False,
+            patch_size: int = 16
+    ):
+        """ Creates a ViT encoder using transformers.ViTModel.
+
+        use_pretrained: If True, uses a pretrained transformer based on the 
+            pretrained_model argument.
+        pretrained: If str, expects the path to a pretrained model or the id of
+            a model on huggingface.co, and ignores the configuration provided in
+            the arguments.
+        """
+        super().__init__()
+        try:
+            from transformers import ViTConfig, ViTModel
+        except ModuleNotFoundError:
+            logger.error(
+                ' transformers is not installed. '
+                'In order to install all image feature dependencies run '
+                'pip install ludwig[image]'
+            )
+            sys.exit(-1)
+
+        img_width = img_width or img_height
+        if not img_width == img_height:
+            raise ValueError(f'img_height and img_width should be identical.')
+        self._input_shape = (in_channels, img_height, img_width)
+
+        if use_pretrained:
+            self.model = ViTModel.from_pretrained(pretrained_model)
+            self.model.train()
```

Comment:
The BertEncoder (see `text_encoders.py`) has a `trainable=True` param the user can set, so that they can disable fine tuning. I think we may as well add something here, too.

Reply:
Done!